variation-commons-core/src/main/java/uk/ac/ebi/eva/commons/core/models/VariantKeyFields.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.ac.ebi.eva.commons.core.models;
+
+import org.apache.commons.lang3.StringUtils;
+
+import static java.lang.Math.max;
+
+/**
+ * Class that stores a Variant coordinates and alleles, normalizing and left aligning them.
+ * <p>
+ * This task comprises 2 steps: removing the trailing bases that are
+ * identical in both alleles, then the leading identical bases.
+ * <p>
+ * It is left aligned because the trailing bases are removed before the leading ones, implying a normalization where
+ * the position is moved the least possible from its original location.
+ */
+public class VariantKeyFields {
```

Comment:
Now that we have a `SubSnpCoreFields` class we should decide on "key" vs "core" for consistency. I think I prefer the latter.